kubernetes/base/leaderelection/leaderelection.py

```diff
@@ -103,6 +107,15 @@ def renew_loop(self):
 
             # failed to renew, return
             return
+    
+    def release(self):
+        lock_status, old_election_record = self.election_config.lock.get(self.election_config.lock.name,
+                                                                        self.election_config.lock.namespace)
+        if lock_status:
```

Comment:
let's check [if the elector is the leader](https://github.com/kubernetes/kubernetes/blob/10066243df4313e28704d3770742f161ceea0112/staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go#L295-L297). We can short-circuit the release method if not